benchmarks/nemo/nemolite2d/manual_versions/psykal_kokkos/time_step_views_kokkos.cpp

```diff
@@ -0,0 +1,616 @@
+
+    // Kokkos needs to be initialized. Since NemoLite2D just has a single
+    // invoke, it is simple to do it here the first time the invoke is
+    // executed. Note that this can not be done for `Kokkos::finalize();`
+    // which is ignored in this implementation. 
+    if(first_time){
+        Kokkos::initialize();
+        first_time = false;
+    }
+
+    // The execution space is given as a preprocessor define when compiling
+    // this file. e.g. `g++ -DEXEC_SPACE=OpenMP time_step_kokkos.cpp -c`
+#if defined (EXEC_SPACE)
+    using execution_space = Kokkos::EXEC_SPACE;
+#else
+    using execution_space = Kokkos::DefaultExecutionSpace;
+#endif
+
+    // Execution policy for a multi-dimensional (2D) iteration space.
+    typedef Kokkos::MDRangePolicy<Kokkos::Rank<2>, execution_space> mdrange_policy;
+
+    // Create 2D View types for the Fields and Grid arrays
+    typedef Kokkos::View<double**> double_2dview;
+    typedef Kokkos::View<int**> int_2dview;
+
+    // Fields
+    double_2dview ssha_t_view("ssha_t", internal_xstop+1, internal_ystop+1);
+    double_2dview sshn_t_view("sshn_t", internal_xstop+1, internal_ystop+1);
+    double_2dview sshn_u_view("sshn_u", internal_xstop+1, internal_ystop+1);
+    double_2dview sshn_v_view("sshn_v", internal_xstop+1, internal_ystop+1);
+    double_2dview hu_view("hu", internal_xstop+1, internal_ystop+1);
+    double_2dview hv_view("hv", internal_xstop+1, internal_ystop+1);
+    double_2dview un_view("un", internal_xstop+1, internal_ystop+1);
+    double_2dview vn_view("vn", internal_xstop+1, internal_ystop+1);
+    double_2dview ua_view("ua", internal_xstop+1, internal_ystop+1);
+    double_2dview ht_view("ht", internal_xstop+1, internal_ystop+1);
+    double_2dview ssha_u_view("ssha_u", internal_xstop+1, internal_ystop+1);
+    double_2dview va_view("va", internal_xstop+1, internal_ystop+1);
+    double_2dview ssha_v_view("ssha_v", internal_xstop+1, internal_ystop+1);
+
+    // Grid
+    int_2dview tmask_view("tmask_v", internal_xstop+1, internal_ystop+1);
+    double_2dview area_t_view("area_t", internal_xstop+1, internal_ystop+1);
+    double_2dview area_u_view("area_u", internal_xstop+1, internal_ystop+1);
+    double_2dview area_v_view("area_v", internal_xstop+1, internal_ystop+1);
+    double_2dview dx_u_view("dx_u", internal_xstop+1, internal_ystop+1);
+    double_2dview dx_v_view("dx_v", internal_xstop+1, internal_ystop+1);
+    double_2dview dx_t_view("dx_t", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_u_view("dy_u", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_v_view("dy_v", internal_xstop+1, internal_ystop+1);
+    double_2dview dy_t_view("dy_t", internal_xstop+1, internal_ystop+1);
+    double_2dview gphiu_view("gphiu", internal_xstop+1, internal_ystop+1);
+    double_2dview gphiv_view("gphiv", internal_xstop+1, internal_ystop+1);
+
+
+    // Create Mirrors (this overlap thew views if the execution device is the host)
```

Comment:
s/thew/the/ I think? Please could you add maybe another line to the comment to explain what the purpose of 'Mirrors' is.

Reply:
Extended the comment

Reply:
Great - thanks.